tests/features/typing_pep563_test.py

```diff
@@ -0,0 +1,148 @@
+            (2, 7),
+            id='not python 3+',
+        ),
+        pytest.param(
+            'from __future__ import annotations\n'
+            'x: Literal["foo", "bar"]\n',
+            (3,),
+            id='Literal',
+        ),
+        pytest.param(
+            'from __future__ import annotations\n'
+            'x = TypeVar("x", "str")\n',
+            (3,),
+            id='TypeVar',
+        ),
+        pytest.param(
+            'from __future__ import annotations\n'
+            'x = cast(x, "str")\n',
+            (3,),
+            id='cast',
+        ),
+        pytest.param(
+            'from __future__ import annotations\n'
+            'x = Annotated["str", "metadata"]\n',
+            (3,),
+            id='Annotated',
+        ),
+    ),
+)
+def test_fix_typing_pep563_noop(s, version):
+    assert _fix_plugins(s, settings=Settings(min_version=version)) == s
+
+
+@pytest.mark.parametrize(
+    ('s', 'expected'),
+    (
+        pytest.param(
+            'from __future__ import annotations\n'
+            'def foo(var: "MyClass") -> "MyClass":\n'
+            '   x: "MyClass"\n',
+
+            'from __future__ import annotations\n'
+            'def foo(var: MyClass) -> MyClass:\n'
+            '   x: MyClass\n',
+
+            id='Annotation',
+        ),
+        pytest.param(
+            'from __future__ import annotations\n'
+            'def foo(\n'
+            '    var0: Tuple["MyClass"],\n'
+            '    *,\n'
+            '    var2: Tuple["MyClass"],\n'
+            ') -> Tuple["MyClass"]:\n'
+            '   x: Tuple["MyClass"]\n',
+
+            'from __future__ import annotations\n'
+            'def foo(\n'
+            '    var0: Tuple[MyClass],\n'
+            '    *,\n'
+            '    var2: Tuple[MyClass],\n'
+            ') -> Tuple[MyClass]:\n'
+            '   x: Tuple[MyClass]\n',
+
+            id='Tuple',
+        ),
+        pytest.param(
+            'from __future__ import annotations\n'
+            'def foo(\n'
+            '   var: Callable[["MyClass"], None]\n'
+            ') -> Callable[["MyClass"], None]:\n'
+            '   x: Callable[["MyClass"], None]\n',
+
+            'from __future__ import annotations\n'
+            'def foo(\n'
+            '   var: Callable[[MyClass], None]\n'
+            ') -> Callable[[MyClass], None]:\n'
+            '   x: Callable[[MyClass], None]\n',
+
+            id='List',
+        ),
+        pytest.param(
+            'from __future__ import annotations\n'
+            'class Foo(NamedTuple):\n'
+            '    x: "MyClass"\n',
+
+            'from __future__ import annotations\n'
+            'class Foo(NamedTuple):\n'
+            '    x: MyClass\n',
+
+            id='NamedTuple',
+        ),
+    ),
+)
+def test_fix_typing_pep563(s, expected):
+    ret = _fix_plugins(s, settings=Settings(min_version=(3, 7)))
+    assert ret == expected
+
+
+@pytest.mark.skipif(
+    sys.version_info.major != 3 or sys.version_info.minor != 8,
```

Comment:
flake8-2020 should really yell at you here -- this logic is buggy don't do that do; `sys.version_info.major < (3, 8)` (think about what happens in 3.9 or 4.0)

also prefer `xfail` to `skip` -- it shouldn't crash and you'll get the coverage for the test without needing a no-cover